ci/travis/determine_tests_to_run.py

```diff
@@ -0,0 +1,93 @@
+# Script used for checking changes for incremental testing cases
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import subprocess
+
+
+def list_changed_files(commit_range):
+    """
+    Returns the list of files changed within commit_range.
+    """
+    command = ["git", "diff", "--name-only", commit_range]
+    proc = subprocess.Popen(
+        command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    out, err = proc.communicate()
+    if proc.returncode != 0:
+        raise RuntimeError("Command {cmdline} failed with code {returncode}, "
+                           "stderr was:\n{stderr}\n".format(
+                               cmdline=command,
+                               returncode=proc.returncode,
+                               stderr=err.decode()))
+
+    return list(filter(None, (s.strip() for s in out.decode().splitlines())))
+
+
+if __name__ == "__main__":
+
```

Comment:
I think we want to test everything in the case that it is a `push` build instead of a `pull_request` build. You can test this by doing `os.environ["TRAVIS_EVENT_TYPE"] == "pull_request"`

Reply:
So we will set everything to 1 if it is a push, otherwise we run our classification for a pull request?

Reply:
Yeah, that's right. We could relax that in the future, but let's start with that. The important thing is that for pushes, we definitely need to build the MacOS and Linux wheels, because those will be stored on S3 and can be downloaded later by people who want to install Ray from a specific git commit.